File: Algorithms/050.Pow.java
public class Solution {
    public double myPow(double x, int n) {
        double result = 1.0;
        if (n == 0)
            return 1.0;
        boolean positive = true;
        long nn = n;
        if (n < 0) {
            positive = false;
            nn = -nn;
        }
        while (nn != 0) {
            if (nn%2 == 1) {
                result *= x;
            }
            x *= x;
            nn /= 2;
        }
        if (positive) {
            return result;
        }
        else {
            return 1/result;
        }
    }
}

File: Algorithms/py/050.Pow.py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        if n == 0:
            return 1.0
        if n < 0:
            return 1/self.myPow(x, -n)
        result = 1.0
        while n:
            if n & 1: 
                result *= x
            n //= 2
            x *= x
        return result